### PR TITLE
chore: add config set LV_WIDGETS_HAS_DEAFULT_VALUE

### PR DESCRIPTION
**Pull request by yushuailong**
### Description of the feature or fix

add config set label default text

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [x] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by fuxiujun** [comment]
Closes #4372

File: lv_conf_template.h [hunk 1]
```diff
@@ -494,6 +494,8 @@
 
 #define LV_USE_WIN        1
 
+#define LV_WIDGETS_HAS_DEAFULT_VALUE  1
```
Comment (kisvegabor) [inline lv_conf_template.h]: Please move it before the first widget.

File: src/widgets/bar/lv_bar.c [hunk 2]
```diff
@@ -231,7 +231,9 @@ static void lv_bar_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
 
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_CHECKABLE);
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_SCROLLABLE);
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_bar_set_value(obj, 0, LV_ANIM_OFF);
```
Comment (kisvegabor) [inline src/widgets/bar/lv_bar.c]: I think we should keep. It just ensures that all internal variables are sete correctly.

File: src/widgets/calendar/lv_calendar.c [hunk 3]
```diff
@@ -282,8 +282,10 @@ static void lv_calendar_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     lv_obj_set_flex_flow(obj, LV_FLEX_FLOW_COLUMN);
     lv_obj_set_flex_grow(calendar->btnm, 1);
 
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_calendar_set_showed_date(obj, calendar->showed_date.year, calendar->showed_date.month);
```
Comment (kisvegabor) [inline src/widgets/calendar/lv_calendar.c:289]: Here we just apply what was set in line 246-252. So we should either remove those or keep these lines too.  I vote for removing line 246-252 too.

File: src/widgets/checkbox/lv_checkbox.c [hunk 4]
```diff
@@ -130,10 +130,12 @@ static void lv_checkbox_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     LV_UNUSED(class_p);
     LV_TRACE_OBJ_CREATE("begin");
 
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_checkbox_t * cb = (lv_checkbox_t *)obj;
-
     cb->txt = (char *)"Check box";
     cb->static_txt = 1;
+#endif
```
Comment (kisvegabor) [inline src/widgets/checkbox/lv_checkbox.c:141]: Please add 
```
#else
    cb->txt = (char *)"";
    cb->static_txt = 1;
#endif
```

File: src/widgets/keyboard/lv_keyboard.c [hunk 5]
```diff
@@ -438,7 +438,9 @@ static void lv_keyboard_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     lv_obj_add_event(obj, lv_keyboard_def_event_cb, LV_EVENT_VALUE_CHANGED, NULL);
     lv_obj_set_style_base_dir(obj, LV_BASE_DIR_LTR, 0);
 
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_keyboard_update_map(obj);
```
Comment (kisvegabor) [inline src/widgets/keyboard/lv_keyboard.c]: Here we just apply `keyboard->mode`. I think we should keep this line ones the "maps" are there.

File: src/widgets/label/lv_label.c [hunk 6]
```diff
@@ -660,9 +660,10 @@ static void lv_label_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
     label->dot_tmp_alloc = 0;
 
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_CLICKABLE);
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_label_set_long_mode(obj, LV_LABEL_LONG_WRAP);
```
Comment (kisvegabor) [inline src/widgets/label/lv_label.c:663]: The long mode should be set in all case to have a sane default state.

File: src/widgets/label/lv_label.c [hunk 7]
```diff
@@ -660,9 +660,10 @@ static void lv_label_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
...
     lv_label_set_text(obj, LV_LABEL_DEFAULT_TEXT);
```
Comment (kisvegabor) [inline src/widgets/label/lv_label.c:664]: Let's add the if around the `LV_LABEL_DEFAULT_TEXT` like:
```c
#if LV_WIDGETS_HAS_DEAFULT_VALUE
   #define LV_LABEL_DEFAULT_TEXT "Text"
#else
   #define LV_LABEL_DEFAULT_TEXT ""
#endif
```

File: src/widgets/spinbox/lv_spinbox.c [hunk 8]
```diff
@@ -354,7 +354,9 @@ static void lv_spinbox_constructor(const lv_obj_class_t * class_p, lv_obj_t * ob
     lv_textarea_set_one_line(obj, true);
     lv_textarea_set_cursor_click_pos(obj, true);
 
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_spinbox_updatevalue(obj);
```
Comment (kisvegabor) [inline src/widgets/spinbox/lv_spinbox.c]: As we need to set some defaults anyway, here I would keep it for consistency to apply the previously set values.

File: src/widgets/textarea/lv_textarea.c [hunk 9]
```diff
@@ -848,7 +848,9 @@ static void lv_textarea_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     lv_obj_add_event(ta->label, label_event_cb, LV_EVENT_ALL, NULL);
     lv_obj_add_flag(obj, LV_OBJ_FLAG_SCROLL_ON_FOCUS);
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_SCROLL_WITH_ARROW);
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_textarea_set_cursor_pos(obj, 0);
```
Comment (kisvegabor) [inline src/widgets/textarea/lv_textarea.c:852]: I think it also makes sense to explicitly set the cursor to the beginning.

Comment (kisvegabor) [inline src/widgets/textarea/lv_textarea.c:852]: I think it also makes sense to explicitly set the cursor to the beginning.

**Review by kisvegabor — COMMENTED** [review]
Thank you! I've added a few comments.

**Pull request merged**